baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -0,0 +1,192 @@
+            with self.baseplate.make_server_span(context, self.name) as span:
+                delivery_info = message.delivery_info
+                span.set_tag("kind", "consumer")
+                span.set_tag("routing_key", delivery_info.get("routing_key", ""))
+                span.set_tag("consumer_tag", delivery_info.get("consumer_tag", ""))
+                span.set_tag("delivery_tag", delivery_info.get("delivery_tag", ""))
+                span.set_tag("exchange", delivery_info.get("exchange", ""))
+                self.handler_fn(context, message)
+        except Exception as exc:
+            logger.exception(
+                "Unhandled error while trying to process a message, re-queueing the message."
+            )
+            message.requeue()
+            if isinstance(exc, AbortError):
+                logger.info("Error signals an abort, terminating.")
+                raise
+        else:
+            message.ack()
+
+
+class KombuQueueConsumerFactory(QueueConsumerFactory):
+    def __init__(
+        self,
+        baseplate: Baseplate,
+        name: str,
+        connection: kombu.Connection,
+        queues: Sequence[kombu.Queue],
+        handler_fn: Handler,
+        health_check_fn: Optional[HealthCheck] = None,
+        serializer: Optional[KombuSerializer] = None,
+    ):
+        self.baseplate = baseplate
+        self.connection = connection
+        self.queues = queues
+        self.name = name
+        self.handler_fn = handler_fn
+        self.health_check_fn = health_check_fn
+        self.serializer = serializer
+
+    @classmethod
+    def new(
+        cls,
+        baseplate: Baseplate,
+        exchange: kombu.Exchange,
+        connection: kombu.Connection,
+        queue_name: str,
+        routing_keys: Sequence[str],
+        handler_fn: Handler,
+        health_check_fn: Optional[HealthCheck] = None,
+        serializer: Optional[KombuSerializer] = None,
+    ) -> "KombuQueueConsumerFactory":
+        queues = []
+        for routing_key in routing_keys:
+            queues.append(kombu.Queue(name=queue_name, exchange=exchange, routing_key=routing_key))
+        return cls(
+            baseplate=baseplate,
+            name=queue_name,
+            connection=connection,
+            queues=queues,
+            handler_fn=handler_fn,
+            health_check_fn=health_check_fn,
+            serializer=serializer,
+        )
+
+    def build_pump_worker(self, work_queue: WorkQueue) -> KombuConsumerWorker:
+        return KombuConsumerWorker(
+            connection=self.connection,
+            queues=self.queues,
+            work_queue=work_queue,
+            serializer=self.serializer,
+        )
+
+    def build_message_handler(self) -> KombuMessageHandler:
+        return KombuMessageHandler(self.baseplate, self.name, self.handler_fn)
+
+    def build_health_checker(self, listener: socket.socket) -> StreamServer:
+        baseplate_configurator = BaseplateConfigurator(self.baseplate)
```

Comment:
do we actually need this whole baseplatey pyramid stack for this? could just make a trivial wsgi app directly?

```python
def healthcheck(environ, start_response):
    start_response("200 OK", [("Content-Type", "text/plain")])
    return ["all good, boss"]

return WSGIServer(listener=listener, application=healthcheck)
```